webrender/src/tiling.rs

```diff
@@ -2616,28 +2612,25 @@ impl FrameBuilder {
                         continue;
                     }
 
-                    match layer.composite_kind {
-                        CompositeKind::None => {}
-                        CompositeKind::Simple(info) => {
-                            let mut prev_task = alpha_task_stack.pop().unwrap();
-                            let item = AlphaRenderItem::Blend(sc_index, current_task.id, info, next_z);
-                            next_z += 1;
-                            prev_task.as_alpha_batch().alpha_items.push(item);
-                            prev_task.children.push(current_task);
-                            current_task = prev_task;
-                        }
-                        CompositeKind::Complex(info) => {
-                            let layer_rect = layer.xf_rect.as_ref().unwrap().bounding_rect;
-                            let readback_task = RenderTask::new_readback(sc_index, layer_rect);
-
-                            let mut prev_task = alpha_task_stack.pop().unwrap();
-                            let item = AlphaRenderItem::Composite(sc_index, readback_task.id, current_task.id, info, next_z);
-                            next_z += 1;
-                            prev_task.as_alpha_batch().alpha_items.push(item);
-                            prev_task.children.push(current_task);
-                            prev_task.children.push(readback_task);
-                            current_task = prev_task;
-                        }
+                    for filter in &layer.composite_ops.filters {
+                        let mut prev_task = alpha_task_stack.pop().unwrap();
+                        let item = AlphaRenderItem::Blend(sc_index, current_task.id, *filter, next_z);
+                        next_z += 1;
```

Comment:
I don't think changing the Z makes sense here, given that we are rendering into the task cache and that's not supposed to use the depth test anyway.